### PR TITLE
DACT-638: Added validation for appointment date

### PR DESCRIPTION
**Pull request by anona-w**
[DACT-638](https://companieshouse.atlassian.net/browse/DACT-638) - Adding validation to appointed_on field

[DACT-638]: https://companieshouse.atlassian.net/browse/DACT-638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java [hunk 1]
```diff
@@ -67,6 +67,8 @@ public ApiErrors validate(HttpServletRequest request, OfficerFilingDto dto, Tran
         // Perform validation
         validateCompanyNotDissolved(request, errorList, companyProfile.get());
         validateAllowedCompanyType(request, errorList, companyProfile.get());
+        validateAppointmentPastOrPresent(request, errorList, dto);
```
Comment (jbestwickch) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java:70]: To keep things consistent, I'd call these form the validateRequiredDtoFields method when appointment date is being checked.

File: src/main/java/uk/gov/companieshouse/officerfiling/api/enumerations/ValidationEnum.java [hunk 2]
```diff
@@ -7,11 +7,14 @@ public enum ValidationEnum {
...
     DIRECTOR_NOT_FOUND("director-not-found"),
     REMOVAL_DATE_IN_PAST("removal-date-in-past"),
+    APPOINTMENT_DATE_IN_PAST("appointment-date-in-past"),
     REMOVAL_DATE_MISSING("removal-date-missing"),
+    APPOINTMENT_DATE_MISSING("appointment-date-missing"),
     DIRECTOR_ALREADY_REMOVED("director-already-removed"),
     REMOVAL_DATE_AFTER_APPOINTMENT_DATE("removal-date-after-appointment-date"),
     REMOVAL_DATE_INVALID("removal-date-invalid"),
     REMOVAL_DATE_AFTER_INCORPORATION_DATE("removal-date-after-incorporation-date"),
+    APPOINTMENT_DATE_AFTER_INCORPORATION_DATE("appointment-date-after-incorporation-date"),
```
[the remark this answers is not in the document]
Comment (jbestwickch) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/enumerations/ValidationEnum.java:17]: Resolved

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java (hunk 1 above)
Comment (jbestwickch) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerAppointmentValidator.java:70]: Anona has pointed out that in TM01 validation we only check length and characters in the requiredDtoFields check. The implemented way fits with this

**Review by jbestwickch — DISMISSED** [review]

**Review by clewis1 — APPROVED** [review]
Looks good

**Pull request merged**